Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -498,8 +498,13 @@ TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)
 			}
 		}
 
-		// Find the entry which hasn't been used for the longest time
-		if (entry->frameCount != FRAMECOUNT_INVALID && entry->frameCount < temp_frameCount)
+		// Find the entry which hasn't been used for the longest time. Some games create
+		// animations by only changing the palette. Keeping all these textures in the cache
+		// improves the performance a lot for these games, so ignore paletted textures with
+		// the same base texture here. Example: Sonic the Fighters (inside Sonic Gems
+		// Collection) loops a 64 frames animation, where only the palette changes
+		if (entry->frameCount != FRAMECOUNT_INVALID && entry->frameCount < temp_frameCount &&
+			!(isPaletteTexture && entry->base_hash == tex_hash))
```

Comment:
I guess you try to workaround here that we shall not invalidate the base texture. But this logic will only stop to invalidate the currently used base texture, so if you have more than one base texture "used", both will get invalidated.
In fact, we just need to set base_texture_entry->frameCount on all usages of the converted textures...